release-content/0.15/release-notes/15281_Add_core_and_alloc_over_std_Lints.md

```diff
@@ -1,4 +1,37 @@
 <!-- Add `core` and `alloc` over `std` Lints -->
 <!-- https://github.com/bevyengine/bevy/pull/15281 -->
 
-<!-- TODO -->
+Bevy is entirely dependent on the standard library, making it challenging to use on embedded, niche
+platforms, and even certain consoles. But what if that _wasn't_ the case?
+
+Bevy's maintainers have undertaken a new initiative to challenge the reliance on the standard
+library, with the eventual goal of providing a [`no_std`](https://docs.rust-embedded.org/book/intro/no-std.html)
+compatible subset of Bevy which could be used on a much wider range of platforms.
+
+The first step of this journey is the addition of a new set of lints:
+
+* `std_instead_of_core`
+* `std_instead_of_alloc`
+* `alloc_instead_of_core`
+
+For those unfamiliar with `no_std` Rust, the standard library, `std`, gets a lot of its functionality
+from two smaller crates, [`core`](https://doc.rust-lang.org/core/) and [`alloc`](https://doc.rust-lang.org/alloc/).
+The `core` crate is available on every Rust target with very few exceptions, providing the
+fundamental infrastructure that the Rust language relies on, such as iterators, `Result`, and many more.
+While the `alloc` crate provides access to allocation-related functionality, such as `Vec`, `Box`,
+and `String`. It's marginally harder to setup on a new platform, but still relatively straightforward.
```

Comment:
I think we can and should explain the goals a little more or at least, the thing that is a must know: Why some modern consoles aren't `no_std` and how this features will help devs of said consoles. I might be wrong, but even consoles like nintendo switch can't really use std, right?

Reply:
I've added some extra explanation to the motivation for this work, with some explanation around Rust's platform support tiers and explicitly calling out the challenges in getting the standard library on modern (NDA) consoles.